### PR TITLE
Improve dev packages

### PR DESCRIPTION
**Pull request by tortmayr**
### Introduce a script for checking copyright header years
This year is coming to an end and with the beginning of 2023 the hustle to ensure that copyright headers are consistently updated starts again. The introduction of the `@eclipse-glsp/cli` package provides us with the opportunity to establish a script for automated checks.

The  `check:Headers` commands validates the copyright headers of the defined fileset to see whether the copyright year (range) is inline with the first and last git modification date. Found violations are printed to the console.
The validation check can be restricted to pending changes and/or the last commit e.g. to validate a commit before creating a PR.

Example:
Execution of a full check with the current glsp-client master shows a couple of outdated headers:
Screenshot from 2022-12-22 09-06-56

Note: I have introduced special handling for copyright headers where the defined start year predates the year of the first commit. Here its easy to get false positives because source code that has been migrated to the repository often has an earlier start here. In addition, some cases of heavy renaming/package restructuring might also lead to false positives.
Currently the following mechanism is in place. If a copy right header with an earlier start year is found then
- check if the file in question was part of the initial contribution. If so consider this to be `ok`
- else mark the file with a warning.

Warnings are not displayed on default but can be optionally enabled. e.g:

Screenshot from 2022-12-22 09-10-44

### Update prettier config
Add plugin to autosort package.jsons files. This ensures that all package.json are sorted consistently and that newly added dependencies or scrips are correctly sorted as well.

### CODE REVIEW

File: dev-packages/cli/README.md [hunk 1]
```diff
@@ -11,6 +11,35 @@ Install `@eclipse-glsp/cli` as a dev dependency in your application.
 yarn add @eclipse-glsp/cli --dev
 ```
 
+## checkHeaders
+
+The `checkHeaders` command can be used to validate the copyright year (range) of license headers.
+It checks for each file (matching the include pattern) whether the defined copyright range is inline with the first and last modification state in the git repository.
```
Comment (martin-fleck-at) [inline dev-packages/cli/README.md:17]:
```suggestion
It checks for each file (matching the include pattern) whether the defined copyright range is in line with the first and last modification date in the git repository.
```

File: dev-packages/cli/src/util/git-util.ts [hunk 2]
```diff
@@ -14,74 +14,159 @@
...
-        return false;
-    }
-    LOGGER.debug('Github CLI is authenticated and ready to use');
-    return true;
+import { getShellConfig } from './command-util';
+
+export function isGitRepository(path?: string): boolean {
+    cdIfPresent(path);
+    return (
+        sh
```
Comment (martin-fleck-at) [inline dev-packages/cli/src/util/git-util.ts:24]: Odd formatting and unnecessary parenthesis, I believe. Was that formatted like this on purpose?

File: dev-packages/cli/src/util/git-util.ts [hunk 3]
```diff
@@ -14,74 +14,159 @@
...
+
+/**
+ * Returns the files that have uncommitted changes (staged, not staged and untracked) of a git repository.
+ * Filepaths are absolute.
+ */
+export function getUncommittedChanges(path?: string): string[] {
     cdIfPresent(path);
-    return sh.exec('git status --porcelain').stdout.trim().length !== 0;
+    return (
+        sh
```
Comment (martin-fleck-at) [inline dev-packages/cli/src/util/git-util.ts:48]: Odd formatting and unnecessary parenthesis, I believe. Was that formatted like this on purpose?

File: dev-packages/cli/src/commands/check-header.ts [hunk 4]
```diff
@@ -0,0 +1,315 @@
...
+    LOGGER.newLine();
+
+    // Adjust results to print based on configured severity level
+    let toPrint = results;
+    if (options.severity === 'error') {
+        toPrint = violations;
+    } else if (options.severity === 'warn') {
+        toPrint = results.filter(result => result.severity !== 'ok');
+    }
+    toPrint.forEach((result, i) => LOGGER.info(`${i}. `, result.file, ':', toPrintMessage(result)));
```
Comment (martin-fleck-at) [inline dev-packages/cli/src/commands/check-header.ts:262]: Not really an issue but the output will show the first line as `0.` so maybe we should print `i+1` here.

File: dev-packages/cli/src/commands/check-header.ts [hunk 5]
```diff
@@ -0,0 +1,315 @@
...
+
+    if (!currentEndYear) {
+        if (currentStartYear === expectedStartYear) {
+            result.violation = 'none';
+            result.severity = 'ok';
+        }
+        return;
+    }
+
+    // Cornercase: For files of the initial contribution the copyright header  predates the first git modification date.
```
Comment (martin-fleck-at) [inline dev-packages/cli/src/commands/check-header.ts:202]:
```suggestion
    // Cornercase: For files of the initial contribution the copyright header predates the first git modification date.
```

**Review by martin-fleck-at — CHANGES_REQUESTED** [review]
Change looks good to me and works great, thank you Tobi! And also thank you for splitting it into multiple commits which made the review much easier ;-) I just have a few nitpicks which you could consider.

File: dev-packages/cli/src/util/git-util.ts (hunk 3 above)
Comment (tortmayr) [reply dev-packages/cli/src/util/git-util.ts:48]: That's prettier autoformatting. I guess the inline comments messes up the formatting. It's a little bit better if I move the comment inside of the map function

File: dev-packages/cli/src/util/git-util.ts (hunk 2 above)
Comment (tortmayr) [reply dev-packages/cli/src/util/git-util.ts:24]: Prettier autoformatting. Not really something I can change here 😉

**Review by martin-fleck-at — APPROVED** [review]
LGTM!

**Pull request merged**